src/Std/Internal/Async/Process.lean

```diff
@@ -0,0 +1,239 @@
+-/
+prelude
+import Std.Time
+import Std.Internal.UV.System
+import Std.Data.HashMap
+
+open Std Time
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+namespace Process
+
+/--
+Represents resource usage statistics for a process or thread.
+All time values are in milliseconds.
+-/
+structure ResourceUsageStats where
+  /--
+  CPU time spent in user mode (milliseconds)
+  -/
+  cpuUserTimeMs : UInt64
+
+  /--
+  CPU time spent in kernel mode (milliseconds)
+  -/
+  cpuSystemTimeMs : UInt64
+
+  /--
+  Peak resident set size (max physical memory usage) in kilobytes
+  -/
+  peakResidentSetSizeKb : UInt64
+
+  /--
+  Size of shared memory segments (kilobytes)
+  -/
+  sharedMemorySizeKb : UInt64
+
+  /--
+  Size of unshared data segment (kilobytes)
+  -/
+  unsharedDataSizeKb : UInt64
+
+  /--
+  Size of unshared stack segment (kilobytes)
+  -/
+  unsharedStackSizeKb : UInt64
+
+  /--
+  Number of minor (soft) page faults (no disk I/O)
+  -/
+  minorPageFaults : UInt64
+
+  /--
+  Number of major (hard) page faults (disk I/O required)
+  -/
+  majorPageFaults : UInt64
+
+  /--
+  Number of swap ins or swap outs
+  -/
+  swapOperations : UInt64
+
+  /--
+  Number of block input operations (disk reads)
+  -/
+  blockInputOps : UInt64
+
+  /--
+  Number of block output operations (disk writes)
+  -/
+  blockOutputOps : UInt64
+
+  /--
+  Number of IPC messages sent
+  -/
+  messagesSent : UInt64
+
+  /--
+  Number of IPC messages received
+  -/
+  messagesReceived : UInt64
+
+  /--
+  Number of signals received
+  -/
+  signalsReceived : UInt64
+
+  /--
+  Number of voluntary context switches (process yielded CPU)
+  -/
+  voluntaryContextSwitches : UInt64
+
+  /--
+  Number of involuntary context switches (process preempted)
+  -/
+  involuntaryContextSwitches : UInt64
+deriving Repr, Inhabited
+
+
+/--
+A process identifier, typically a numeric ID like in UNIX (e.g. 1001).
+-/
+structure PId where
+  /--
+  The numeric process ID.
+  -/
+  toUInt64 : UInt64
+deriving Inhabited, DecidableEq, Ord
+
+instance : Repr PId where
+  reprPrec u := Repr.addAppParen ("PId.mk " ++ repr u.toUInt64)
+
+/--
+Gets the title of the current process.
+-/
+@[inline]
+def getProcessTitle : IO String :=
+  UV.System.getProcessTitle
+
+/--
+Sets the title of the current process.
+-/
+@[inline]
+def setProcessTitle (title : String) : IO Unit :=
+  UV.System.setProcessTitle title
+
+/--
+Gets the current process id.
+-/
+@[inline]
+def getId : IO PId :=
+  PId.mk <$> UV.System.osGetPid
+
+/--
+Gets the current process parent id.
+-/
+@[inline]
+def getParentId : IO PId :=
+  PId.mk <$> UV.System.osGetPpid
+
+/--
+Gets the current working directory.
+-/
+@[inline]
+def getCwd : IO System.FilePath :=
+  UV.System.cwd
+
+/--
+Changes the current working directory to a new one.
+-/
+@[inline]
+def setCwd (path : System.FilePath) : IO Unit :=
+  UV.System.chdir path.toString
+
+/--
+Gets the scheduling priority of the current process.
```

Comment:
It says `current` in the docstring, but you take a pid. Similar for `setPriority` below.